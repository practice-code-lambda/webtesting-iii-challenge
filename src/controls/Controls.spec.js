import React from 'react';
import renderer from 'react-test-renderer';
// import { render, fireEvent } from '@testing-library/react';
import Controls from './Controls'
import { render, fireEvent } from 'react-testing-library';

describe('<Controls/>', () => {
    it('should take a snapshot', () => {
        const snap = renderer.create(<Controls />).toJSON();
        //Jest
        expect(snap).toMatchSnapshot()
    })
    it('should execute a function', () => {
        const mock = jest.fn()
        const { getByTestId } = render(<Controls toggleClosed={mock} />)

        const button = getByTestId(/toggleClosed/i)
        // console.log(button)
        fireEvent.click(button)

        expect(mock).toHaveBeenCalled()







    })
})

