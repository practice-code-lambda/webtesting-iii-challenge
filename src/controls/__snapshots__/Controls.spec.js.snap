// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls/> should take a snapshot 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    data-testid="togglelock"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
    data-testid="toggleClosed"
  >
    Close Gate
  </button>
</div>
`;
